cmd/deploy/deploy.go

```diff
@@ -513,6 +513,17 @@ func getDefaultTimeout() time.Duration {
 
 func GetDeployer(ctx context.Context, manifest *model.Manifest, opts *Options, builder *buildv2.OktetoBuilder, cmapHandler configMapHandler) (deployerInterface, error) {
 
+	isRunningRemotelyEnvVar := utils.LoadBoolean(constants.OktetoForceRemote)
+
+	if opts.Manifest != nil && opts.Manifest.Deploy != nil {
+		if !isRunningRemotelyEnvVar || opts.Manifest.Deploy.Remote == true {
+			opts.RunInRemote = true
+		}
+	} else {
+		err := oktetoErrors.ErrCouldNotInferAnyManifest
+		return nil, err
+	}
+
 	if isRemoteDeployer(opts.RunInRemote, opts.Manifest.Deploy.Image) {
```

Comment:
I think the calculation of whether to run remotely or not has become quite complicated. Could we summarize it in this function? cc @jLopezbarb 
````go
func runInRemote(opts *Options) bool {

	// already in remote so we need to deploy locally
	if utils.LoadBoolean(constants.OktetoDeployRemote) {
		return false
	}

	if opts.RunInRemote {
		return true
	}

	if opts.Manifest != nil && opts.Manifest.Deploy != nil {
		if opts.Manifest.Deploy.Image != "" || opts.Manifest.Deploy.Remote {
			return true
		}
	}

	if utils.LoadBoolean(constants.OktetoForceRemote) {
		return true
	}

	return false
}

func GetDeployer(ctx context.Context, manifest *model.Manifest, opts *Options, builder *buildv2.OktetoBuilder, cmapHandler configMapHandler) (deployerInterface, error) {

	if runInRemote(opts) {
		// run remote
		oktetoLog.Info("Deploying remotely...")
		return newRemoteDeployer(builder), nil
	}

	// run local
	oktetoLog.Info("Deploying locally...")

	deployer, err := newLocalDeployer(ctx, opts, cmapHandler)
	if err != nil {
		return nil, fmt.Errorf("could not initialize local deploy command: %w", err)
	}
	return deployer, nil
}
````